pkg/model/config/config.go

```diff
@@ -194,31 +307,43 @@ func (c Config) Marshal() ([]byte, error) {
 
 	// Append extra fields to put them at the config's bottom.
 	if len(c.Plugins) != 0 {
-		// Unless the project version is v2 which does not support a plugins field.
-		if cfg.IsV2() {
-			return nil, fmt.Errorf("error marshalling project configuration: plugin field found for v2")
-		}
-
 		pluginConfigBytes, err := yaml.Marshal(Config{Plugins: c.Plugins})
 		if err != nil {
-			return nil, fmt.Errorf("error marshalling project configuration extra fields: %v", err)
+			return nil, marshalError{err}
 		}
+
 		content = append(content, pluginConfigBytes...)
 	}
 
 	return content, nil
 }
 
-// Unmarshal unmarshals the bytes of a Config into c.
+type unmarshalError struct {
+	error
+}
+
+func (e unmarshalError) Error() string {
+	return fmt.Sprintf("error unmarshalling project configuration: %s", e.error.Error())
+}
+
+func (e unmarshalError) Unwrap() error {
+	return e.error
+}
+
+// Unmarshal unmarshalls the bytes of a Config into c.
 func (c *Config) Unmarshal(b []byte) error {
 	if err := yaml.UnmarshalStrict(b, c); err != nil {
-		return fmt.Errorf("error unmarshalling project configuration: %v", err)
+		return unmarshalError{err}
 	}
 
-	// Project versions < v3 do not support a plugins field.
-	if !c.IsV3() {
-		c.Plugins = nil
+	// To simplify the project configuration files we have omitted some fields that could be restored, do it now
+	c.restore()
```

Comment:
I do not think that we should do it. Change the data that is obtained from the PROJECT file introduces a complexity that is not required at all.

Reply:
Restore allows things like only storing irregular plurals. If they are regular they are omitted at marshalling and restored at unmarshalling.

Reply:
I'm ok with doing simplify/restore, and current (un)marshal code already does something similar for plugin configs.

Reply:
I am not ok with that because it reduces the maintainability. Imagine how harder it can become to be keep maintained if we have these if's for many versions. Also, it shows hurts a few concepts.